src/main/daml/Daml/Finance/Instrument/Swap/Fpml/Instrument.daml

```diff
@@ -89,12 +97,19 @@ template Instrument
             let l = last calculationSchedule
             assertMsg "The adjusted terminationDate must match the adjusted end date of the last period" $ l.adjustedEndDate == terminationDateAdjusted
 
+            -- If there is an fxLinkedNotionalSchedule, look up the base notional from a different swap leg
+            let
+              constantNotionalScheduleReference = case c.fxLinkedNotionalSchedule of
+                None -> None
+                Some fxl -> getNotionalFromOtherSwapStream fxl.constantNotionalScheduleReference
+
             -- Calculate the claims from the current leg (either fix or float)
-            if isSome c.fixedRateSchedule && isNone c.floatingRateCalculation then calculateClaimsFromFixSwapStream s calculationPeriodicSchedule calculationSchedule paymentSchedule useAdjustedDatesForDcf issuerPaysLeg notional currency
-            else if isNone c.fixedRateSchedule && isSome c.floatingRateCalculation then calculateClaimsFromFloatingSwapStream s calculationPeriodicSchedule calculationSchedule paymentSchedule useAdjustedDatesForDcf issuerPaysLeg notional currency issuer calendarDataProvider
+            if isSome c.fixedRateSchedule && isNone c.floatingRateCalculation then calculateClaimsFromFixSwapStream s calculationPeriodicSchedule calculationSchedule paymentSchedule useAdjustedDatesForDcf issuerPaysLeg ccy
```

Comment:
pattern matching to the helpr?

Reply:
Good point, thanks! I did it as a tuple pattern matching to catch the interesting combinations and only have one error message.